common/src/main/java/fr/esrf/TangoApi/DeviceProxy.java

```diff
@@ -2020,6 +2080,10 @@ private static void checkDuplication(String[] list, String orig) throws DevFaile
      */
     //==========================================================================
     protected void finalize() {
```

Comment:
Finalize is deprecated in Java 9 and is discouraged even before. Should use `AutoCloseable` and implement `close` mrthod.